### DISCUSSION TITLE
Yellowish result after converting cr2/raf/dng to tiff/jpg using Magick.NET-Q16-AnyCPU.

### DISCUSSION THREAD

**Discussion by B-V-A** [dlemstra/Magick.NET#886] [Help]
Hello, everybody!
Thank you a lot to those who contributes for a great ImageMagick!
My name is Vladimir.
I have some issue while converting raw files (cr2, raf) to tiff or jpg.
Here is a simple code, nothing special:

            using (MagickImage magickimage = new MagickImage(@"D:\Образцы raw файлов\IMG_0028.cr2"))
            {
                magickimage.Format = MagickFormat.Tiff;
                magickimage.SetCompression(CompressionMethod.NoCompression);
                magickimage.BitDepth(16);
                magickimage.Write(@"D:\Образцы raw файлов\IMG_0029.tif");
            }

I always get yellowish result… 

I have tried different versions of Magick.NET-Q16-AnyCPU (7.23.3, 7.22.3, 7.14.4) and settings but they cannot remove this yellowish color.
What could be a problem?
I have attached raw files and also original and result snapshots:
[Files.zip](https://github.com/dlemstra/Magick.NET/files/6203784/Files.zip)

Thanks for any help in advance!

P.S. Console application ImageMagick from: https://imagemagick.org also produces yellowish result.
If I use app called RawTherapee, then everything is ok.

**Comment by dlemstra** [comment]
This is happening due to auto brightness in the libraw library that is being used. You can disable that like this:

```C#
var readSettings = new MagickReadSettings
{
    Defines = new DngReadDefines
    {
        DisableAutoBrightness = true,
    },
};

using (MagickImage magickimage = new MagickImage(@"D:\Образцы raw файлов\IMG_0028.cr2", readSettings))
```

**Comment by BFNOC** [comment]
I had the same problem, but i use like this:

```csharp
var readSettings = new MagickReadSettings
{
    Defines = new DngReadDefines
    {
        UseAutoWhitebalance = true,  //you can try which one is better.
        UseCameraWhitebalance = true,   //you can try which one is better.
    },
};
using (var image = new MagickImage("your CR2 file", readSettings))
```
Then the color of the output picture is normal
